sdk/common/src/main/java/io/opentelemetry/sdk/resources/internal/EntityUtil.java

```diff
@@ -0,0 +1,295 @@
+  }
+
+  /** Appends a new collection of entities on to the end of the list of entities. */
+  public static final ResourceBuilder addAllEntity(ResourceBuilder rb, Collection<Entity> e) {
+    try {
+      Method method = ResourceBuilder.class.getDeclaredMethod("addAll", Collection.class);
+      if (method != null) {
+        method.setAccessible(true);
+        method.invoke(rb, e);
+      }
+    } catch (NoSuchMethodException nme) {
+      logger.log(Level.WARNING, "Attempting to use entities with unsupported resource", nme);
+    } catch (IllegalAccessException iae) {
+      logger.log(Level.WARNING, "Attempting to use entities with unsupported resource", iae);
+    } catch (InvocationTargetException ite) {
+      logger.log(Level.WARNING, "Attempting to use entities with unsupported resource", ite);
+    }
+    return rb;
+  }
+
+  /**
+   * Returns a collectoion of associated entities.
+   *
+   * @return a collection of entities.
+   */
+  @SuppressWarnings("unchecked")
+  public static final Collection<Entity> getEntities(Resource r) {
+    try {
+      Method method = Resource.class.getDeclaredMethod("getEntities");
+      if (method != null) {
+        method.setAccessible(true);
+        return (Collection<Entity>) method.invoke(r);
+      }
+    } catch (NoSuchMethodException nme) {
+      logger.log(Level.WARNING, "Attempting to use entities with unsupported resource", nme);
+    } catch (IllegalAccessException iae) {
+      logger.log(Level.WARNING, "Attempting to use entities with unsupported resource", iae);
+    } catch (InvocationTargetException ite) {
+      logger.log(Level.WARNING, "Attempting to use entities with unsupported resource", ite);
+    }
+    return Collections.emptyList();
+  }
+
+  /**
+   * Returns a map of attributes that describe the resource, not associated with entites.
+   *
+   * @return a map of attributes.
+   */
+  public static final Attributes getRawAttributes(Resource r) {
+    try {
+      Method method = Resource.class.getDeclaredMethod("getRawAttributes");
+      if (method != null) {
+        method.setAccessible(true);
+        return (Attributes) method.invoke(r);
+      }
+    } catch (NoSuchMethodException nme) {
+      logger.log(Level.WARNING, "Attempting to use entities with unsupported resource", nme);
+    } catch (IllegalAccessException iae) {
+      logger.log(Level.WARNING, "Attempting to use entities with unsupported resource", iae);
+    } catch (InvocationTargetException ite) {
+      logger.log(Level.WARNING, "Attempting to use entities with unsupported resource", ite);
+    }
+    return Attributes.empty();
+  }
+
+  /** Returns true if any entity in the collection has the attribute key, in id or description. */
+  public static final <T> boolean hasAttributeKey(
+      Collection<Entity> entities, AttributeKey<T> key) {
+    return entities.stream()
+        .anyMatch(
+            e -> e.getId().asMap().containsKey(key) || e.getDescription().asMap().containsKey(key));
```

Comment:
There's a couple of places where we the code calls `Attributes.asMap().containsKey(key)`. Its much more efficient to call `Attributes.get(key) != null`.